### PR TITLE
Agency/Provider adjustments for Modes

### PR DESCRIPTION
**Pull request by schnuerle**
## Explain pull request

Refinement of Agency and Provider Unification with Modes.

## Is this a breaking change

* Yes

## Impacted Spec

Which spec(s) will this pull request impact?

* `agency`
* `provider`

## Additional context

N/A

### CODE REVIEW

File: data-types.md
Comment (marie-x) [inline data-types.md:117]: I thought that by definition there could only be one journey at a time?

File: data-types.md
Comment (marie-x) [inline data-types.md:198]: I always hated having the `provider_name` as well as the `provider_id` because of the redundancy.  I will defer to the will of the group but I want to make my case that it's a stupid-simple lookup on a small table to get the provider name.

File: data-types.md
Comment (marie-x) [inline data-types.md:201]: Same argument here. We have a device registry, and the whole point was to NOT include redundant info over and over and over.

File: data-types.md
Comment (marie-x) [inline data-types.md:218]: All cost fields should go in the fare info, not here.

File: data-types.md
Comment (marie-x) [inline data-types.md:217]: I thought I left this in? Got lost along the way I guess?

**Review by marie-x — CHANGES_REQUESTED** [review]

File: data-types.md
Comment (schnuerle) [reply data-types.md:117]: Ah I see, yes. I was thinking somehow if it spanned multiple modes or something. You are right, should remain singular.

File: data-types.md
Comment (schnuerle) [reply data-types.md:218]: Well just like there are some basic trip fields that span all modes, there are some basic fare fields that also span all modes.  Hence keeping these core ones here.

File: data-types.md
Comment (schnuerle) [reply data-types.md:201]: I guess since other data in /vehicles now like vehicle_attributes, propulsion_types, and accessibility_options, then fields like vehicle_id and vehicle_type could also be removed from /trips.

File: data-types.md
Comment (schnuerle) [reply data-types.md:198]: Yes agreed, my mistake. We have talked about this in WG meetings and were waiting for the next major breaking release.

**Comment by schnuerle** [comment]
@marie-x Where is [the vehicle status object](https://github.com/openmobilityfoundation/mobility-data-specification/blob/feature-smarter-scooters/data-types.md#vehicle-status) used elsewhere in the spec?

**Comment by schnuerle** [comment]
@marie-x I think we need to have the Reports object move to data types, and a reports endpoint added to Agency, for full symmetry. What do you think?

**Comment by marie-x** [comment]
> 

Makes sense, I just hadn't given it any thought!

**Comment by marie-x** [comment]
> Where is [the vehicle status object](https://github.com/openmobilityfoundation/mobility-data-specification/blob/feature-smarter-scooters/data-types.md#vehicle-status) used elsewhere in the spec?

Oops, that's an oversight - it's supposed to be in the fleet-status payload in Provider.

**Comment by schnuerle** [comment]
> > Where is [the vehicle status object](https://github.com/openmobilityfoundation/mobility-data-specification/blob/feature-smarter-scooters/data-types.md#vehicle-status) used elsewhere in the spec?
> 
> Oops, that's an oversight - it's supposed to be in the fleet-status payload in Provider.

Can you mark where on the provider page it's supposed to go, using the line-item comment here in the PR?

File: agency/README.md
Comment (schnuerle) [inline agency/README.md:301]: @marie-x I added Reports to Agency. Check this out for any errors and that it makes sense to you technically.

File: provider/README.md
Comment (schnuerle) [inline provider/README.md:183]: @marie-x Added individual vehicle lookup option to Provider, like in Agency. Does this work and make sense?

File: agency/README.md
Comment (schnuerle) [inline agency/README.md:20]: @marie-x Can we rename the Agency vehicle events endpoint from /vehicles/events to just /events? And /vehicles/telemetry to just /telemetry? Would provide symmetry with Provider endpoint names. Adding vehicles seems redundant since it is implied and is described in the docs, just like in Provider, and we don't for example say /vehicle/trips. Is there any reason to keep it for events and telemetry now?

File: provider/README.md
Comment (marie-x) [reply provider/README.md:183]: yep!

File: agency/README.md
Comment (marie-x) [reply agency/README.md:301]: looks good!

**Review by marie-x — APPROVED** [review]

**Pull request merged**
